Source/Examples/ExampleLibrary/Misc/ToolTipsExamples.cs

```diff
@@ -0,0 +1,197 @@
+        public static PlotModel SimpleToolTipsOnAnnotations()
+        {
+            var model = new PlotModel {
+                Title = "Hover me to see a tooltip",
+                Subtitle = "Me too",
+                TitleToolTip = "Our tooltip"
+            };
+
+            model.Axes.Add(new LinearAxis { Position = AxisPosition.Bottom });
+
+            model.Axes.Add(new LinearAxis { Position = AxisPosition.Left });
+
+            model.Annotations.Add(new LineAnnotation { Slope = 0.1, Intercept = 1, Text = "LineAnnotation", ToolTip = "This is a tooltip for the LineAnnotation" });
+
+            model.Annotations.Add(new RectangleAnnotation { MinimumX = 20, MaximumX = 70, MinimumY = 10, MaximumY = 40, TextRotation = 10, Text = "RectangleAnnotation", ToolTip = "This is a tooltip for the RectangleAnnotation", Fill = OxyColor.FromAColor(99, OxyColors.Blue), Stroke = OxyColors.Black, StrokeThickness = 2 });
+
+            model.Annotations.Add(new EllipseAnnotation { X = 20, Y = 60, Width = 20, Height = 15, Text = "EllipseAnnotation", ToolTip = "This is a tooltip for the EllipseAnnotation", TextRotation = 10, Fill = OxyColor.FromAColor(99, OxyColors.Green), Stroke = OxyColors.Black, StrokeThickness = 2 });
+            
+            model.Annotations.Add(new PointAnnotation { X = 50, Y = 50, Text = "P1", ToolTip = "This is a tooltip for the PointAnnotation" });
+
+            model.Annotations.Add(new ArrowAnnotation { StartPoint = new DataPoint(8, 4), EndPoint = new DataPoint(0, 0), Color = OxyColors.Green, Text = "ArrowAnnotation", ToolTip = "This is a tooltip for the ArrowAnnotation" });
+
+            model.Annotations.Add(new TextAnnotation { TextPosition = new DataPoint(60, 60), Text = "TextAnnotation", ToolTip = "This is a tooltip for the TextAnnotation" });
+
+            return model;
+        }
+
+        [Example("Tooltips with frequent refreshes")]
+        public static PlotModel ToolTipsWithFrequentRefreshes()
+        {
+            var plot = new PlotModel()
+            {
+                IsLegendVisible = true,
+                Title = "Mouse events are shown here"
+            };
+
+            var fs = new FunctionSeries(x => Math.Sin(x), 0, 10, 0.001);
+            plot.Series.Add(fs);
+
+            fs = new FunctionSeries(x => Math.Cos(x), 0, 10, 0.001);
+            plot.Series.Add(fs);
+
+            var ra = new RectangleAnnotation()
+            {
+                MinimumX = 0.5,
+                MinimumY = -0.8,
+                MaximumX = 9.5,
+                MaximumY = -0.6,
+                ToolTip = "Test",
+                Text = "Hover the rectangle to see the fixed tooltip.\n" +
+                    "Hover the flashing series to see a new random number each time you hover it.",
+            };
+            plot.Annotations.Add(ra);
+
+            var rnd = new Random();
+            _ = WaitThenChangeColorAndToolTip(plot, rnd, fs);
+
+            plot.MouseDown += (s, e) =>
```

Comment:
@VisualMelon I am not sure how to do this. My first impression is to use the extension method `IController.BindMouseDown` with a parameter of type `DelegateViewCommand` but I do not know how code-gen handles this, and, more important to me, I do not know what `IController` method to use instead of the model's `MouseUp` event.

Thank you.

Reply:
I'm note sure, I've never used them myself. However, take a look at `ManipulatorBase`: you should be able to bind a `MouseDownGesture`, and implement the started/delta/completed events to handle the entire interaction.

Reply:
@VisualMelon I posted my unsuccessful attempt on this in a gist. The relevant lines with comments are [here](https://gist.github.com/silviubogan/b9eeac18ac4db3ccf0c7b5e534df2f31#file-using-oxyplot-controller-events-in-examples-cs-L62-L64). I think I will try more tomorrow. (I think that I must also find how to use the `OxyMouseDownGesture` class). It would be great if you could help me more. Thank you for all the help.

Reply:
Manipulators are, I believe, a transient thing. Lots of examples in `PlotCommands.cs` (e.g. [`PanAt'](https://github.com/oxyplot/oxyplot/blob/13c686502ebba28eeca1011ad77eb25ea5788723/Source/OxyPlot/PlotController/PlotCommands.cs#L28). I'd expect the code to be more like this:

    plot.PlotView.ActualController.BindMouseDown(OxyMouseButton.Left, OxyModifierKeys.None, new DelegateViewCommand<OxyMouseDownEventArgs>((v, c, e) =>
    {
        plot.PlotView.ActualController.AddMouseManipulator(plot.PlotView, new CustomMouseManipulator(plot.PlotView, rnd), e);
    }));

However, this can never work, because we just created the model, so it can't have a plotview or controller, so my gut feeling is that answer to @objorke question is that it is not possible. I think I'm going to have to try to get my head around #111...